pdfbox/src/test/java/org/apache/pdfbox/pdmodel/font/TestFontEmbedding.java

```diff
@@ -371,4 +376,116 @@ void testReuseEmbeddedSubsettedFont() throws IOException
             assertEquals(text1 + " " + text2, extractedText.trim());
         }
     }
+
+    private class TrueTypeEmbedderTester extends TrueTypeEmbedder {
+        /**
+         * Common functionality for testing the TrueTypeFontEmbedder
+         *
+         */
+        TrueTypeEmbedderTester(PDDocument document, COSDictionary dict, TrueTypeFont ttf, boolean embedSubset) throws IOException {
+            super(document, dict, ttf, embedSubset);
+        }
+
+        @Override
+        protected void buildSubset(InputStream ttfSubset, String tag, Map<Integer, Integer> gidToCid) throws IOException {
+            // no-op.  Need to define method to extend abstract class, but
+            // this method is not currently needed for testing
+        }
+    }
+
+    /**
+     * Test that we validate embedding permissions properly for all legal permissions combinations
+     *
+     * @throws IOException
+     */
+    @Test
+    void testIsEmbeddingPermittedMultipleVersons() throws IOException
```

Comment:
This PR is branched off of the trunk branch.  I think this is essentially the pdfbox 3.0.x prerelease code.  The test framework used in 2.0.x and 3.0.x is quite a bit different.  I think they use different libraries and frameworks for unit testing.  These tests are compatible with the 3.0.x code but not the 2.0.x code.  I think I'll need to have a talk with the pdfbox maintainers and see how they want to handle that.  For our purposes, we care about the 2.0.x code (at least for now).